src/components/AllComponents.js

```diff
@@ -34,40 +31,35 @@ function AllComponents() {
   const [suggestion, setSuggestion] = useState([]);
   const [webOfTeam, setWebOfTeam] = useState([]);
   
-  const { loading, data } = useQuery(QUERY_CHARACTERS);
-  const allCharacters = data?.characters || [];
-  
   const [getUserData, { loading:loading3, data:data3 }] = useLazyQuery(GET_USERDATA);
   const [getUserCharactersById, { loading:loading4, data:data4 }] = useLazyQuery(GET_USERCHARACTERSBYID);
 
   const [onLoadGetUserCharacters, setOnLoadGetUserCharacters] = useState([]);
   const [userCharacters, setUserCharacters] = useState([]);
-  
+
   useEffect(() => {
     console.log('Not sure if logged in')
 
     const auth = async () => {
       if (Auth.loggedIn()) {
         const username = await Auth.getProfile().data.username
-        getUserData({
+        const result = await getUserData({
           variables: {
             username: username
           },
-        }).then((result) => {
-          setOnLoadGetUserCharacters(result.data.me.savedCharacters)
-          console.log(result);
         });
-        
+        setOnLoadGetUserCharacters(result.data.me.savedCharacters)
+        console.log(result);        
       }
       
     }
     auth();
-    // if(Auth.loggedIn()) {
-    //   console.log("logged in")
-    //   console.log(username);
-    // }
   },[])
 
+  useEffect(() => {
```

Comment:
This is just moved from line 81 (and removed the unused setCharacters). Still needed to initialize filteredCharacters once allCharacters loads